service-python/featuretoggle/src/lib/queues.py

```diff
@@ -0,0 +1,54 @@
+import json
+import logging
+from datetime import datetime
+
+import pytz
+
+from . import main, utils
+from .redis_client import RedisClient
+from .settings import queue_config, redis_config
+
+EXCHANGE = queue_config["exchange_name"]
+TOGGLE_QUEUE = queue_config["toggle_queue_name"]
+
+
+def feature_toggle_request_callback(channel, method, properties, body):
+
+    redis_client = RedisClient(redis_config)
+    binding_key = method.routing_key
+    message = json.loads(body.decode("utf-8"))
+    logging.info(f" [x] {binding_key}: Received message.")
+
+    try:
+        response = main.report_feature_toggles(message)
+        redis_client.publish("features", items=utils.create_features_list())
+        redis_client.publish("features_timestamp", pytz.utc.localize(datetime.now()))
```

Comment:
I'm under the impression that this microservice gets called whenever another container wants a current list of feature toggles. If so, what is the purpose of publishing the complete list to these two Redis channels every time?

Reply:
Ideally the plan is for the main app to call this once and propagate through other routes as needed. Complete list + timestamp published in preparation for future plans where things may be toggled after deployment (so the list could potentially be updated) and timestamp to verify against last check. I believe the java implementation will requery every N minutes or so (#802 / (earlier discussion in #245)

Another container could call this separately from the overall app as well, if necessary

Reply:
> I'm under the impression that this microservice gets called whenever another container wants a current list of feature toggles. If so, what is the purpose of publishing the complete list to these two Redis channels every time?

I'd like for the microservice to support 2 usage scenarios:
- "pull": a client calls the microservice for all features toggles (as you mentioned)
- "push": a client subscribes to changes of feature toggles